feat: add flutter replay docs (#11611)
getsentry/sentry-docs: docs/platforms/flutter/session-replay/index.mdx

---
title: Set Up Session Replay
sidebar_order: 5500
notSupported:
description: "Learn how to enable the Beta of Mobile Session Replay in your app."
---

<Note>

Mobile support for Session Replay is in Beta. Features available in Beta are still work-in-progress and may have bugs. We recognize the irony.

If you have any questions, feedback or would like to report a bug, please open a [GitHub issue](https://github.com/getsentry/sentry-dart/issues/new?template=BUG_REPORT.yml) with a link to a relevant replay in Sentry if possible.

</Note>

[Session Replay](/product/explore/session-replay/) helps you get to the root cause of an error or latency issue faster by providing you with a reproduction of what was happening in the user's device before, during, and after the issue. You can rewind and replay your application's state and see key user interactions, like taps, swipes, network requests, and console entries, in a single UI.

By default, our Session Replay SDK masks all text content, images, and user input, giving you heightened confidence that no sensitive data will leave the device. To learn more, see [product docs](/product/explore/session-replay/).

> ## Pre-requisites
> 
> Make sure your Sentry Flutter SDK version is at least 8.9.0, which is required for Session Replay. 
> You can update your `pubspec.yaml` to the matching version:

Reviewer: ```suggestion
## Prerequisites

Sentry Flutter SDK version 8.9.0 and up. You can update your `pubspec.yaml` by running the below:
```

Author: > by running the below:

that doesn't work here as below is not an executable script. It's a snippet of the configuration file and they already would have the `sentry_flutter` dependency added - they just need to make sure they have the minimum version with replay available.

```yaml
dependencies:
  sentry_flutter: ^8.9.0
```

> ## Set Up

Reviewer: ```suggestion
## Setup
```

> To set up the integration, add the following to your Sentry initialization.

Reviewer: ```suggestion
To set up the integration, add the following to your Sentry initialization:
```

```dart
await SentryFlutter.init(
  (options) {
    ...
    options.experimental.replay.sessionSampleRate = 1.0;
    options.experimental.replay.onErrorSampleRate = 1.0;
  },
  appRunner: () => runApp(MyApp()),
);
```

## Verify

While you're testing, we recommend that you set <PlatformIdentifier name="replays-session-sample-rate" /> to `1.0`. This ensures that every user session will be sent to Sentry.

Once testing is complete, **we recommend lowering this value in production**. We still recommend keeping <PlatformIdentifier name="replays-on-error-sample-rate" /> set to `1.0`.

## Sampling

Sampling allows you to control how much of your website's traffic will result in a Session Replay. There are two sample rates you can adjust to get the replays relevant to you:

> 1. <PlatformIdentifier name="replays-session-sample-rate" /> - The sample rate for
>    replays that begin recording immediately and last the entirety of the user's session.

Reviewer: ```suggestion
1. <PlatformIdentifier name="replays-session-sample-rate" /> - The sample rate for
   replays that begin recording immediately and last the entirety of a user's session.
```

2. <PlatformIdentifier name="replays-on-error-sample-rate" /> - The sample rate for
   replays that are recorded when an error happens. This type of replay will record
   up to a minute of events prior to the error and continue recording until the session
   ends.

> Sampling begins as soon as a session starts. <PlatformIdentifier name="replays-session-sample-rate" /> is evaluated first. If it's sampled, the replay recording will begin. Otherwise, <PlatformIdentifier name="replays-on-error-sample-rate" /> is evaluated and if it's sampled, the integration will begin buffering the replay and will only upload it to Sentry if an error occurs. The remainder of the replay will behave similarly to a whole-session replay.
> 

Reviewer: ```suggestion
Sampling starts as soon as a session begins. The <PlatformIdentifier name="replays-session-sample-rate" /> is then evaluated. If the session is sampled, replay recording will start immediately. If not, <PlatformIdentifier name="replays-on-error-sample-rate" /> will be evaluated. If the session is sampled at this point, the replay will be buffered and will only be uploaded to Sentry if an error occurs. 

```

Author: FYI same code is in other platform's session-replay/index.mdx files

## Privacy

The SDK is recording and aggressively redacting (masking) all text and images, according to the configuration in `options.experimental.replay`.
You can tune this and add custom masking rules to fit your needs. For example, you can explicitly mask or unmask widgets by type,
or you can even have a callback to decide whether a specific widget instance should be masked:

```dart
  options.experimental.replay.mask<IconButton>();
  options.experimental.replay.unmask<Image>();
  options.experimental.replay.maskCallback<Text>(
      (Element element, Text widget) =>
          (widget.data?.contains('secret') ?? false)
              ? SentryMaskingDecision.mask
              : SentryMaskingDecision.continueProcessing);
```

> You can find more details in respective methods' code documentation.

Reviewer: ```suggestion
You can find more details in the documentation for each method.
```

<Note>

> If you encounter any data not being redacted with the default settings, please let us know through a [GitHub issue](https://github.com/getsentry/sentry-dart/issues/new?template=BUG_REPORT.yml).
> 

Reviewer: ```suggestion
If you find that data isn't being redacted with the default settings, please let us know by creating a [GitHub issue](https://github.com/getsentry/sentry-dart/issues/new?template=BUG_REPORT.yml).

```
